x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -678,6 +685,47 @@ private List<Attribute> resolveUsingColumns(List<Attribute> cols, List<Attribute
             return resolved;
         }
 
+        private LogicalPlan resolveInsist(Insist insist, List<Attribute> childrenOutput, IndexResolution indexResolution) {
+            Attribute resolvedCol = maybeResolveAttribute(insist.getInsistIdentifier(), childrenOutput);
+            // Field isn't mapped anywhere.
+            if (resolvedCol instanceof UnresolvedAttribute) {
+                return pushdownInsist(insist, attrs -> attrs.add(insistKeyword(insist)));
+            }
```

Comment:
1. Modifying/eliminating the nodes happens in the optimizers - the analyzer needs to preserve its content for reporting purposes in case the query is invalid.
2. Instead convert the UnresolvedAttribute into a UnmappedAttribute (which is resolved) or FieldAttribute pointing to the UnmappedEsField which is later picked by the verifier (which checks that it should always have a relationship to push it into otherwise throws an error).